src/core/timer.c

```diff
@@ -169,6 +170,48 @@ static int timer_setup_persistent(Timer *t) {
         return 0;
 }
 
+static int timer_set_fixed_delay(Timer *t) {
+        struct prefix {
+                sd_id128_t machine_id;
+                bool is_system_manager;
+                uid_t uid;
+        } *prefix;
+
+        static const uint8_t hash_key[] = {
+                0x51, 0x0a, 0xdb, 0x76, 0x29, 0x51, 0x42, 0xc2,
+                0x80, 0x35, 0xea, 0xe6, 0x8e, 0x3a, 0x37, 0xbd
+        };
+
+        _cleanup_free_ uint8_t *fixed_delay_data = NULL;
+        size_t dlen, id_strlen;
+        uint64_t hash;
+        int r;
+
+        assert(t);
+
+        id_strlen = strlen(UNIT(t)->id);
+        dlen = sizeof(struct prefix) + id_strlen;
```

Comment:
i'd always put an array with an unspcified length at the end of the struct, and then use offsetof() to determine its offset, to which to add the string length.

(that said, don't bother with the struct, just use siphash_compress() individually for the four pieces of information to hash, see comment below)